HHS/Head-Start-TTADP
# [TTAHUB-251] Warn of simultaneous editing of reports

## Description of change

- Installed the "ws" and "redis" packages on the backend to handle websockets and the node redis connection respectively.
- Use redis & websockets on the backend to create a pub sub channel 
- Frontend uses websockets when on an activity report to report its location within the report
- Displays an alert based on the messages coming down the pipe

## How to test
- Update packages if necessary
- You need to have Redis running
- Create a report. If you look at your network tab, you should have a websocket connection that publishes data on save.
- If you want to view the message, you can edit ```frontend/src/components/SocketAlert.js:27``` to not filter out the current user. You'll need to have two tabs on the same page, and you'll have to wait for the save event before the alert appears.
- [Mockup](https://ocio-jira.acf.hhs.gov/browse/TTAHUB-720)

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-431
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-251


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
